include/uncertainty_planning_core/uncertainty_contact_planning.hpp

```diff
@@ -141,6 +140,100 @@ class UncertaintyPolicyExecutionResult
   }
 };
 
+#if UNCERTAINTY_PLANNING_CORE__SUPPORTED_ROS_VERSION == 2
+struct ExecutionTimeLimit {
+  ExecutionTimeLimit(double _secs)
+    : secs(_secs), clock(std::make_shared<rclcpp::Clock>())
+  {
+  }
+
+  ExecutionTimeLimit(double _secs, rclcpp::Clock::SharedPtr _clock)
+    : secs(_secs), clock(_clock)
+  {
+  }
+
+  double secs;
+  rclcpp::Clock::SharedPtr clock;
+};
+
+std::ostream& operator<<(
+    std::ostream& strm, const ExecutionTimeLimit& time_limit)
+{
+  return strm << time_limit.secs;
+}
+#elif UNCERTAINTY_PLANNING_CORE__SUPPORTED_ROS_VERSION == 1
+using ExecutionTimeLimit = double;
+#endif
+
+class ExecutionTimer {
+public:
+#if UNCERTAINTY_PLANNING_CORE__SUPPORTED_ROS_VERSION == 2
```

Comment:
Let's change this to a single constructor
```
explicit ExecutionTimer(const ExecutionTimeLimit& time_limit) : time_limit_(time_limit) {}
```

Reply:
See https://github.com/calderpg/uncertainty_planning_core/commit/2d3a145375854e0caadd7204655a6f32761e023c.